the-collab-lab/tcl-27-smart-shopping-list
# Add List and Add an Item views and routes.

Co-authored-by: Yolanda Haynes <yhaynes@giyo.us>

## Description

This PR adds three new components: 
1. A List view 
2. Add an Item view
3. Nav bar at the bottom of the page

### A List view

Created a basic "list" view component. This component will be used to display a shopping list. 

### Add an Item view

Created a basic "add an item" component. This component will be used to add items to a shopping list. 

### Nav bar at the bottom of the page

Created a basic navigation component. This component allows the user to toggle between the "list" and "add an item" views. We used [React Router](https://reactrouter.com/web/guides/quick-start) to add the routes for each view and used [Switch](https://reactrouter.com/web/api/Switch) to toggle between the two views. 







## [Acceptance Criteria](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/2)

:white_check_mark: `react-router-dom` has been added as a project dependency
:white_check_mark: Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
:white_check_mark: When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
:white_check_mark: Whichever view is selected, the corresponding link should display in bold text

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |


## Testing Steps / QA Criteria

From your terminal, pull down this branch with `git pull origin tc-yh-add-view-navigation` and check that branch out with `git checkout tc-yh-add-view-navigation`.
Then `npm install` or `yarn` to install the newly added dependency locally and `npm start` or `yarn start` to launch the app.
Click on the List and Add an Item links at the bottom of the page. Make sure that the route and the view changes when you click each link. Make sure that whichever view is selected, the corresponding link is displayed in bold text.
